### PR TITLE
feat(go-sdk): init go sdk

### PR DESCRIPTION
**Pull request by qsliu2017**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**
add a no-op go sql driver

### CODE REVIEW

**Comment by aceforeverd** [comment]
[ospp - Golang SDK](https://trello.com/c/7aquDamb/118-ospp-golang-sdk)

**Comment by aceforeverd** [comment]
@qsliu2017 maybe you need some example code and test cases first ( readme and how to use it ) ? 
We can checking the interface first.

File: go/conn.go [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+// ExecContext implements driver.ExecerContext.
+func (c *conn) ExecContext(ctx context.Context, query string, args []interfaces.NamedValue) (interfaces.Result, error) {
+	return &result{}, c.query(ctx, query)
+}
+
+// QueryContext implements driver.QueryerContext.
+func (c *conn) QueryContext(ctx context.Context, query string, args []interfaces.NamedValue) (interfaces.Rows, error) {
+	return &rows{}, c.query(ctx, query)
+}
```
Comment (aceforeverd) [inline go/conn.go:290]: we require impl of `QueryRowContext`

**Comment by aceforeverd** [comment]
you may consider data types in go. go has the [ColumnType](https://pkg.go.dev/database/sql#ColumnType)

File: .github/workflows/sdk.yml [hunk 2]
```diff
@@ -343,9 +342,43 @@ jobs:
...
+          OPENMLDB_NS_PORT: 6527
+        run: |
+          echo "CREATE DATABASE test_db;" | ./openmldb/bin/openmldb --host=$OPENMLDB_NS_HOST --port=$OPENMLDB_NS_PORT
+
+      - name: go test
+        env:
+          OPENMLDB_APISERVER_HOST: 127.0.0.1
+          OPENMLDB_APISERVER_PORT: 8080
+        working-directory: go
+        run: go test ./...
```
Comment (aceforeverd) [inline .github/workflows/sdk.yml]: the job do not upload test results

File: .github/workflows/sdk.yml [hunk 3]
```diff
@@ -343,9 +342,43 @@ jobs:
...
+
+      - uses: actions/setup-go@v3
+        with:
+          go-version: 1.18
+
+      - name: build openmldb
+        run: make build install
+
+      - name: start server
+        run: ./openmldb/bin/start-standalone.sh
```
Comment (aceforeverd) [inline .github/workflows/sdk.yml:362]: we should test on cluster mode

File: go/driver.go [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+	}
+
+	if u.Scheme != "openmldb" && u.Scheme != "" {
+		return "", "", fmt.Errorf("invalid URL: unknown schema '%s'", u.Scheme)
+	}
+
+	p := strings.Split(strings.TrimLeft(u.Path, "/"), "/")
+
+	if len(p) == 0 {
+		return "", "", fmt.Errorf("invalid URL: DB name not found")
```
Comment (aceforeverd) [inline go/driver.go]: does api server support connection without db ?

File: go/conn.go [hunk 5]
```diff
@@ -0,0 +1,262 @@
...
+
+	return &r, nil
+}
+
+func (c *conn) query(ctx context.Context, sql string, parameters ...interfaces.Value) (rows interfaces.Rows, err error) {
+	if c.closed {
+		return nil, interfaces.ErrBadConn
+	}
+
+	reqBody, err := parseReqToJson("offsync", sql, parameters...)
```
Comment (aceforeverd) [inline go/conn.go]: user should able to control query is online or offline, via dsn ?

File: .github/workflows/sdk.yml [hunk 6]
```diff
@@ -343,9 +342,43 @@ jobs:
           TWINE_USERNAME: ${{ secrets.PYPI_USERNAME }}
           TWINE_PASSWORD: ${{ secrets.PYPI_PASSWORD }}
 
+  go-sdk:
+    runs-on: ubuntu-latest
+    container:
+      image: ghcr.io/4paradigm/hybridsql:latest
```
Comment (aceforeverd) [inline .github/workflows/sdk.yml:348]: the job only need compile openmldb. create an environment variable `OPENMLDB_BUILD_TARGET` and set to `openmldb`

File: go/driver.go (hunk 4 above)
Comment (qsliu2017) [reply go/driver.go]: I think no, at least for now. because apiserver has no rest endpoint that does not need db

**Review by vagetablechicken — APPROVED** [review]
Plz add a doc about how to use go sdk later

File: .github/workflows/sdk.yml [hunk 7]
```diff
@@ -343,9 +342,63 @@ jobs:
...
+    env:
+      OPENMLDB_BUILD_TARGET: "openmldb thirdparty-fast"
```
Comment (aceforeverd) [inline .github/workflows/sdk.yml]: `thirdparty-fast` not needed here, already depended by `build` target

**Comment by aceforeverd** [comment]
@qsliu2017 https://github.com/4paradigm/OpenMLDB/actions/runs/3102510202/jobs/5024874140#step:9:43 tests failed ?

**Comment by qsliu2017** [comment]
> @qsliu2017 [4paradigm/OpenMLDB/actions/runs/3102510202/jobs/5024874140#step:9:43](https://github.com/4paradigm/OpenMLDB/actions/runs/3102510202/jobs/5024874140#step:9:43) tests failed ?

addressed

**Review by aceforeverd — APPROVED** [review]

**Pull request merged**